puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/test/test_foreman.groovy

```diff
@@ -0,0 +1,199 @@
+foreman_branch = 'develop'
+
+pipeline {
+    agent { label 'fast' }
+
+    options {
+        timestamps()
+        timeout(time: 2, unit: 'HOURS')
+        ansiColor('xterm')
+    }
+
+    stages {
+        stage('Test Matrix') {
+            parallel {
+                stage('ruby-2.5-postgres') {
+                    agent { label 'fast' }
+                    environment {
+                        RUBY = '2.5'
+                        GEMSET = 'ruby-2.5-postgres'
+                    }
+                    stages {
+                        stage("setup-2.5-postgres") {
+                            steps {
+                                git url: 'https://github.com/theforeman/foreman', branch: foreman_branch
+                                configureRVM(env.RUBY, env.GEMSET)
+                                databaseFile(gemset(env.GEMSET))
+                                configureDatabase(env.RUBY, env.GEMSET)
+                                withRVM(['npm install'], env.RUBY, env.GEMSET)
+                            }
+                        }
+                        stage("unit-tests-2.5-postgres") {
+                            steps {
+                                withRVM(['bundle exec rake jenkins:unit TESTOPTS="-v" --trace'], env.RUBY, env.GEMSET)
+                            }
+                        }
+                        stage("integration-tests-2.5-postgres") {
+                            steps {
+                                withRVM(['bundle exec rake jenkins:integration TESTOPTS="-v" --trace'], env.RUBY, env.GEMSET)
+                            }
+                        }
+                        stage("assets-precompile-2.5-postgres") {
```

Comment:
IIRC `jenkins:integration` also calls `assets:precompile` so it might be good to swap the order (setup -> unit-tests -> assets-precompile -> integration-tests)

Reply:
These are all in parallel and thus independent, ordering doesn't matter.